engines/python/setup/djl_python/rolling_batch/rolling_batch.py

```diff
@@ -168,34 +171,36 @@ def inference(self, input_data, parameters):
         """
         pass
 
-    def get_new_requests(self, input_data, parameters, batch_size):
-        new_requests = []
-        pending_req_len = len(self.pending_requests)
-        if batch_size > pending_req_len:
-            for i in range(pending_req_len, batch_size):
-                data = input_data[i]
-                params = parameters[i] if i < len(parameters) else {}
-                request = Request(self.req_id_counter, data, params)
-                self.pending_requests.append(request)
-                new_requests.append(request)
-                self.req_id_counter += 1
+    def get_new_requests(self, input_data, parameters):
+        init_batch_size = len(self.pending_requests)
+        for data, params in zip(input_data, parameters):
+            request = Request(self.req_id_counter, data,
+                              params if params else {})
+            self.pending_requests.append(request)
```

Comment:
QQ: Here all the incoming input_data is new requests here? I thought in the front end we have a list of all the requests that are not yet completed.

Reply:
Got it, Yeah so front end has a list of all unfinished requests and append new requests to the list. It would send the whole input str for the new requests, but empty strings for already pending requests. instead of None, like this 
`"inputs": [""]`

Ref: https://github.com/deepjavalibrary/djl-serving/blob/master/engines/python/src/main/java/ai/djl/python/engine/RollingBatch.java#L200

Try with this and check whether it works. 
```
for _ in range(50):
    result = rolling_batch.inference(["", ""], [{}, {}])
    for i, res in enumerate(result):
        output_all[i].append(res['data']) 
    print(' '.join(out) for out in output_all)
```